Automattic/wp-calypso
# Pattern Library: Fix incorrect pattern permalink generated from search results page

Fixes https://github.com/Automattic/dotcom-forge/issues/6608

## Proposed Changes

This PR fixes a bug in the logic that generates pattern permalinks. Previously, if a user was searching all categories and copied a permalink to a page layout, `/layouts` wouldn't get included in the URL.

This happened because `getPatternPermalink` received the pattern type as an argument based on the current view context. Search happens across both regular patterns and page layouts since #89199, meaning there is no view context while searching.

## Testing Instructions



1. Navigate to `/patterns`
2. Search for "about"
3. Copy the permalink for the second result `Store: Description and product categories`
4. Open a new tab and navigate to the copied URL
5. Ensure that you end up on the correct pattern
6. Navigate to `/patterns/about`
7. Copy the permalink for the first pattern in that category
8. Open a new tab and navigate to the copied URL
9. Ensure that you end up on the correct pattern
